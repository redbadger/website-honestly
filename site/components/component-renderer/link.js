// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
