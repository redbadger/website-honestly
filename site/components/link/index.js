// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
