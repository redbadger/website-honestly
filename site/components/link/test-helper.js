// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
