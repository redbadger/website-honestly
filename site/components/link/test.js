// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
