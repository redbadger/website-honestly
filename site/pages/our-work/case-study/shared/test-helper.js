// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
