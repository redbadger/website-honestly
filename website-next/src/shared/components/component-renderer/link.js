import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
