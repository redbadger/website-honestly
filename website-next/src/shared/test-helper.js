import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
