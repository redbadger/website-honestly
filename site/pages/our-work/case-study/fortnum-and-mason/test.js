import React from 'react';
import { render } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import CaseStudyFortnumAndMason from './index';

jest.mock(
  '../shared/what-to-read-next',
  () => require.requireActual('../shared/test-helper').default,
);

describe('site/pages/our-work/case-study/fortnum-and-mason/', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <CaseStudyFortnumAndMason />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
