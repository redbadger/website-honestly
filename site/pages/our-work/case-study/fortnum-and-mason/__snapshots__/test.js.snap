// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/fortnum-and-mason/ renders correctly 1`] = `
<div
  class="caseStudy"
>
  <div
    class="header"
  >
    <picture
      class="picture"
    >
      <source
        media="largeScreen"
        srcset="header-large.jpg"
      />
      <source
        media="mediumScreen"
        srcset="header-medium.jpg"
      />
      <img
        alt="Fortnum and Mason logo embellished by gold scrolls"
        class="img header__image"
        srcset="header-small.jpg"
      />
    </picture>
  </div>
  <div
    class="content"
  >
    <div
      class="content__wrapper"
    >
      <div
        class="content__mainTitleWrapper"
      >
        <h1
          class="content__mainTitle"
        >
          Fortnum & Mason’s new, elegant website increases revenue and conversion rates
        </h1>
      </div>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Let's make things better.
        </span>
        Scaling the site, decreasing drop-outs and bringing the in-store experience, online
      </h2>
      <p
        class="content__paragraph"
      >
        When you think of Fortnum & Mason, the iconic eau de nil colour, the beautiful store layout, amazing customer service and the incredibly high quality of products, springs to mind. This unrivalled reputation, held over 300 years, has always been led by the store experience. But with more and more shoppers heading online to make purchases from over 130 different countries, Fortnum’s realised their digital experience didn’t match the service delivered by the physical stores, and something needed to change, quickly.
      </p>
      <div
        class="content__float"
      >
        <img
          alt="A tablet showing Fortnum & Mason tea website"
          class="tabletImage"
          src="tablet-restaurant.png"
        />
      </div>
      <p
        class="content__paragraph"
      >
        The old site had an inconsistent design, was inflexible, difficult to update and the underlying technology was nearing the end of its life. The out-dated technology meant that high levels of customers were dropping out at certain stages in the buying process. For example, the presentation of delivery options was confusing and if one element of a customer’s order was unavailable, it could result in the entire order not going out for delivery.
      </p>
      <p
        class="content__paragraph"
      >
        In addition, this site needed to solve the above problems as well as handle complex orders and delivery to more than 130 countries. A scalable, highly flexible new site was needed, and very quickly to avoid more lost revenue.
      </p>
      <img
        alt="Fortnum & Mason website on desktop, tablet and mobile devices"
        class="devicesImage"
        src="devices.png"
      />
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Do the right thing. Do the thing right.
        </span>
        Embracing organisational change and implementing cutting-edge technology
      </h2>
      <p
        class="content__paragraph"
      >
        We began with a two-day hackathon to produce a minimum viable product (MVP) to demonstrate our thinking for the new website; a ‘show not tell’ approach.
      </p>
      <div
        class="contactBox"
      >
        <h2
          class="contactBox__heading"
        >
          Project in mind?
        </h2>
        <button
          class="contactBox__button"
          type="button"
        >
          Tell us more
        </button>
      </div>
      <p
        class="content__paragraph"
      >
        Working in a collaborative, Lean way with Fortnum & Mason, our talented design team and knowledgeable tech team, incorporated a progressive selection of open source technology including Spree Commerce, Wombat and Facebook’s React.
      </p>
      <p
        class="content__paragraph"
      >
        We also designed, implemented and managed the Cloud-based architecture upon which the new platform runs. This design allows for a continuous delivery model, meaning that tiny incremental changes are constantly tested and released to production multiple times a day (instead of a few times a year), helping to bid farewell to the age of big bang releases.
      </p>
      <div
        class="content__float"
      >
        <div
          class="quotation"
        >
          <div
            class="quotation__text"
          >
            <blockquote
              class="quotation__quote"
            >
              Since the site went live it has had incredible results in increased revenues and conversion rates.
            </blockquote>
          </div>
          <div
            class="quotation__author"
          >
            <div
              class="quotation__author__wrapper"
            >
              <div
                class="quotation__author__name"
              >
                — Zia Zareem-Slade
              </div>
              <div
                class="quotation__author__title"
              >
                Customer Experience Director, Fortnum & Mason
              </div>
            </div>
          </div>
        </div>
      </div>
      <p
        class="content__paragraph"
      >
        The process and technology had the customer at the heart of the project but the key to the success was engaging Fortnum’s customers to help drive the direction of the design and development of the site by allowing them to provide feedback directly to the team. Throughout the project, our design team ran guerrilla testing in the flagship store in Piccadilly to get customer feedback on the look and feel of the site. The in-store historian was also brought into the project to ensure the new design was as aligned to the brand’s heritage as possible.
      </p>
      <p
        class="content__paragraph"
      >
        Once launched, the beta site ran in parallel to the old site to avoid disruption and a number of Fortnum’s customers were gradually switched over to the new site over a two-month period. New site visitors were able to provide feedback to enable further customer-centric amends and builds.
      </p>
      <p
        class="content__paragraph"
      >
        These changes meant that when the full go-live launch took place the site already had features and processes in place that made sense to the user, not just the business. This approach also meant that overall the project has achieved a complete re-platform and design from concept through to delivery, in just eight months and for under £1.3 milllion.
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Creating lasting change.
        </span>
        Increasing conversion, sales and mobile visits and winning multiple awards
      </h2>
      <div
        class="content__float content__float--left content__float--content-hack"
      >
        <img
          alt="A tablet showing Fortnum & Mason delivery site"
          class="tabletImage"
          src="tablet-checkout.png"
        />
      </div>
      <p
        class="content__paragraph"
      >
        The new Badger-created site was one of the only fully responsive retail sites that was live at the time, which ensures that the customer experience is equally superlative no matter what device is used. The site is also built to be scalable and flexible enough to support the company’s future ambitions and peaks of online traffic, such as Christmas.
      </p>
      <p
        class="content__paragraph"
      >
        As well as being beautiful to look at with a consistent design and the famous eau de nil colour prominent throughout, the new site is fast, slick and easy to navigate - and truly responsive, delivering optimal layouts across every view port.
      </p>
      <p
        class="content__paragraph"
      >
        The site reinforces the brand experience that Fortnum & Mason customers expect and since it went fully live it has delivered incredible sales uplift (38% over the Christmas 2015 period) and dramatically increased conversion.
      </p>
      <p
        class="content__paragraph"
      >
        The results in the first few weeks were striking:
      </p>
      <div
        class="Card listBoxContainer"
      >
        <div
          class="listBox listBox"
        >
          <ul>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Improved conversion in the first week of release 
              </div>
              <div>
                +14%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                 Increase in average order value in the first week of release
              </div>
              <div>
                +5%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Total sales achieving growth year on year
              </div>
              <div>
                +52%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Increase in mobile visits up year on year
              </div>
              <div>
                +77%
              </div>
            </li>
          </ul>
        </div>
      </div>
      <p
        class="content__paragraph"
      >
        The drop-out rate at the delivery options stage was previously 18.8% and has now been reduced to 12.9% (based figures from 17–25th February 2014 vs 2015.)
      </p>
      <p
        class="content__paragraph"
      >
        Other key stats year on year*:
      </p>
      <div
        class="Card listBoxContainer"
      >
        <div
          class="listBox listBox"
        >
          <ul>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Mobile Conversion rate 
              </div>
              <div>
                +57%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Tablet visits
              </div>
              <div>
                +30%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Tablet Conversion
              </div>
              <div>
                +28%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                New customers conversion
              </div>
              <div>
                +15%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Reduction in calls to the Customer Service team
              </div>
              <div>
                -18%
              </div>
            </li>
          </ul>
        </div>
      </div>
      <p
        class="content__paragraph"
      >
        * Statistics are based on launch week to 8th March and same date range last year.
      </p>
      <p
        class="content__paragraph"
      >
        With the site now being much easier to navigate there has been an increase in sales of specific items, for example a 78% increase in grocery sales.
      </p>
      <p
        class="content__paragraph"
      >
        Further to this, fortnumandmason.com won a BIMA Award for ‘Best Web Design & Build’ and an eCommerce Award for ‘Best eCommerce New Design or Re-Design (Under 1 Year)’ in 2015. The project was also nominated for the UK Digital Experience Awards in the Retail sector award and as ‘Best Use of Website to Build Your Brand’. It was also nominated at the Retail Week Tech and Ecommerce Awards in the ‘Best Customer Experience’ category.
      </p>
      <div
        class="videoContainer"
      >
        <span>
          <div
            id="fm-case-study-video"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work"
    >
      See more of our work
    </a>
  </div>
  <section
    class="contactUsContainer"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>
</div>
`;
